openmsupply/application-manager-server
# #690F fix review assignments duplicated for level 1 after LOQ is submitted

Fixes  front-end https://github.com/openmsupply/application-manager-web-app/issues/690

### Changes
- Created new method `getExistingReviewAssignments` in **action_generate_review_assignment** to get existing ReviewAssignments
- When generating (or updating existing) ReviewAssignment will set `isLocked` to true if other ReviewAssignment already existing is Assigned, and keep the same status and isLocked on existing ReviewAssignments.
- Fix tests in `generateReviewAssignments` to be receiving new `isLocked` as part of resulted reviewAssignments
- Fix problem with `updateReviewStatuses` to be aligned with current application/assignments

### Manual testing
- Use application ABC456 logged in with user **Reviewer1**
- Start review and DECLINE one response and submit the LOQ to applicant
- Log in with user **Valerio** and fix the response then re-submit
- Log in again as **Reviewer1** to Re-review and APPROVE all then CONFORM and submit
- Now the application is on the next stage. No other reviewAssignments have been created for other reviewer in stage 1

Check in the master to see the bug happening - you can check the logs to see new reviewAssignments created after re-submitting the application with LOQ back to the reviewer.
